mlflow/tracking/_tracking_service/client.py

```diff
@@ -219,30 +223,70 @@ def get_trace(self, request_id) -> Trace:
         trace_data = self._download_trace_data(trace_info)
         return Trace(trace_info, trace_data)
 
-    def search_traces(
+    def _search_traces(
         self,
         experiment_ids: List[str],
         filter_string: Optional[str] = None,
         max_results: int = SEARCH_TRACES_DEFAULT_MAX_RESULTS,
         order_by: Optional[List[str]] = None,
         page_token: Optional[str] = None,
-    ) -> PagedList[Trace]:
-        trace_infos, token = self.store.search_traces(
+    ):
+        return self.store.search_traces(
             experiment_ids=experiment_ids,
             filter_string=filter_string,
             max_results=max_results,
             order_by=order_by,
             page_token=page_token,
         )
+
+    def search_traces(
+        self,
+        experiment_ids: List[str],
+        filter_string: Optional[str] = None,
+        max_results: int = SEARCH_TRACES_DEFAULT_MAX_RESULTS,
+        order_by: Optional[List[str]] = None,
+        page_token: Optional[str] = None,
+    ) -> PagedList[Trace]:
+        def download_trace_data(trace_info: TraceInfo) -> Optional[Trace]:
+            """
+            Downloads the trace data for the given trace_info and returns a Trace object.
+            If the download fails (e.g., the trace data is missing or corrupted), returns None.
```

Comment:
@dbczumar Should we mention this in the `search_traces` docstring?

Reply:
I don't think so, because it's an invalid trace.

Reply:
got it!